pipeline_dp/dp_engine.py

```diff
@@ -110,3 +114,20 @@ def unnest_cross_partition_bound_sampled_per_key(pid_pk_v):
         return self._ops.flat_map(col,
                                   unnest_cross_partition_bound_sampled_per_key,
                                   "Unnest")
+    
+    def _select_private_partitions(self, col):
+        """
+        Selects and publishes private partitions.
+
+        Args:
+            col: collection, with types for each element: 
+                (partition_key, privacy_id_count, value)
+        
+        Returns:
+            collection of elements (partition_key, data)
+        """
+        def filter_fn(row):
+            return self._partition_selection_strategy.should_keep(row[1])
+
+        filtered_values = self._ops.filter(col, filter_fn)
+        return self._ops.group_by_key(filtered_values, lambda tup: (tup[0], tup[2]))
```

Comment:
In this line it's more correct to use `_ops.map` instead of group_by_key, but since input data format update only `filter` is needed.